### PR TITLE
FIX: Correctly check SDCFlows registry of ``IntendedFor`` files

### PR DESCRIPTION
**Pull request by oesteban**
Follows up on #140 

EDIT 12/17/2020 --

## Summary

This PR explores the new SDCFlows API to apply fieldmaps. Although the control over the available fieldmaps is almost none and many issues are still to be solved, this is a second step towards the integration.

Now, the CircleCI reports contain SDC results reportlets for ds001771 - one of them looks decent, the other needs some investigation.

This dataset contains two fake `_fieldmap` images - these are the two with the weird mask (the problem is that SDCFlows expects a magnitude image, but receives an EPI, because these are fake fieldmaps derived from the TOPUP solution).

### CODE REVIEW

**Comment by oesteban** [comment]
Failing for the same root cause of nipreps/niworkflows#158.

**Comment by josephmje** [comment]
> This dataset contains two fake `_fieldmap` images - these are the two with the weird mask 
> (the problem is that SDCFlows expects a magnitude image, but receives an EPI, because these 
> are fake fieldmaps derived from the TOPUP solution).

Ha! I tested this out earlier and was wondering why the fieldmaps looked the same as the TOPUP solution.

File: dmriprep/config/reports-spec.yml [hunk 1]
```diff
@@ -27,10 +27,30 @@ sections:
...
-  ordering: session,run
+  ordering: session,run,fmapid
   reportlets:
+  - bids: {datatype: figures, desc: mapped, suffix: fieldmap}
+    caption: Inhomogeneities of the <em>B<sub>0</sub></em> field introduce (oftentimes severe) spatial distortions
+      along the phase-encoding direction of the image. Some scanners produce a <em>B<sub>0</sub></em>
+      mapping of the field, using Spiral Echo Imaging (SEI) or postprocessing a "phase-difference"
+      acquisition. The plot below shows an anatomical "magnitude" reference and the corresponding
+      fieldmap.
+    description: Hover on the panels with the mouse pointer to also visualize the intensity of the
```
Comment (josephmje) [inline dmriprep/config/reports-spec.yml:38]:
```suggestion
    description: Hover over the panels with the mouse pointer to also visualize the intensity of the
```

File: dmriprep/config/reports-spec.yml [hunk 2]
```diff
@@ -27,10 +27,30 @@ sections:
...
+      inhomogeneity of the field in Hertz.
```
Comment (josephmje) [inline dmriprep/config/reports-spec.yml:39]:
```suggestion
      field inhomogeneity in Hertz.
```

File: dmriprep/config/reports-spec.yml [hunk 3]
```diff
@@ -27,10 +27,30 @@ sections:
...
+    static: false
+    subtitle: "Susceptibility-derived Distortion Correction (SDC): field inhomogeneity estimation"
+  - bids: {datatype: figures, desc: phasediff, suffix: fieldmap}
+    caption: Inhomogeneities of the <em>B<sub>0</sub></em> field introduce (oftentimes severe) spatial distortions
+      along the phase-encoding direction of the image. A Gradient-Recalled Echo scheme for the
```
Comment (josephmje) [inline dmriprep/config/reports-spec.yml:44]:
```suggestion
      along the phase-encoding direction of the image. A Gradient-Recalled Echo (GRE) scheme for the 
```

File: dmriprep/config/reports-spec.yml [hunk 4]
```diff
@@ -27,10 +27,30 @@ sections:
...
+      inhomogeneity of the field in Hertz.
+    static: false
+    subtitle: "Susceptibility-derived Distortion Correction (SDC): field inhomogeneity estimation"
+  - bids: {datatype: figures, desc: phasediff, suffix: fieldmap}
+    caption: Inhomogeneities of the <em>B<sub>0</sub></em> field introduce (oftentimes severe) spatial distortions
+      along the phase-encoding direction of the image. A Gradient-Recalled Echo scheme for the 
+      mapping of the <em>B<sub>0</sub></em> inhomogeneities by subtracting the phase maps obtained at
+      two subsequent echoes. The plot below shows an anatomical "magnitude" reference and the corresponding
+      fieldmap.
+    description: Hover on the panels with the mouse pointer to also visualize the intensity of the
```
Comment (josephmje) [inline dmriprep/config/reports-spec.yml:48]:
```suggestion
    description: Hover over the panels with the mouse pointer to also visualize the intensity of the
```

File: dmriprep/config/reports-spec.yml [hunk 5]
```diff
@@ -27,10 +27,30 @@ sections:
...
+      inhomogeneity of the field in Hertz.
```
Comment (josephmje) [inline dmriprep/config/reports-spec.yml:49]:
```suggestion
      field inhomogeneity in Hertz.
```

File: dmriprep/config/reports-spec.yml [hunk 6]
```diff
@@ -44,11 +64,16 @@ sections:
...
-      The red contour shows the brain mask calculated using this reference b=0.
-    subtitle: Reference b=0 and brain mask
+    caption: Average <em>b=0</em> that serves for reference in early preprocessing steps.
+    descriptions: The reference <em>b=0</em> is obtained as the voxel-wise median across
+      all <em>b=0</em> found in the dataset, after accounting for signal drift.
+      The red contour shows the brain mask calculated using this reference <em>b=0</em>.
+    subtitle: Reference <em>b=0</em> and brain mask
+  - bids: {datatype: figures, desc: sdc, suffix: dwi}
+    caption: Susceptibility distortions correction (SDC).
+    description: The reportlet shows a <em>b=0</em> reference <em>before</em> and <em>after</em> correction.
```
Comment (josephmje) [inline dmriprep/config/reports-spec.yml:74]:
```suggestion
    description: The reportlet shows a <em>b=0</em> reference <em>before</em> and <em>after</em> distortion correction.
```

**Comment by oesteban** [comment]
> Ha! I tested this out earlier and was wondering why the fieldmaps looked the same as the TOPUP solution.

Wonder no more. I think this is something that @edickie spotted a long while ago.

**Review by josephmje — COMMENTED** [review]
Looks good to me! I just made a few small suggestions for the wording of the reportlet. Waiting for the tests to run to see what you mean by the pre-processing of the other fieldmap.

**Pull request merged**